scripts/uni_tag_allocation.py

```diff
@@ -0,0 +1,97 @@
+import os
+import sys
+import datetime
+from kytos.core.db import Mongo
+from collections import defaultdict
+
+def update_database(mongo: Mongo):
+    db = mongo.client[mongo.db_name]
+
+    circuits = db.evcs.find({},
+        {"uni_a": 1, "uni_z": 1}
+    )
+    intf_update = defaultdict(list)
+    for circuit in circuits:
+        tag_a = circuit["uni_a"].get("tag")
+        tag_z = circuit["uni_z"].get("tag")
+        if tag_a:
+            intf_a = circuit["uni_a"]["interface_id"]
+            intf_update[intf_a].append(tag_a["value"])
+        if tag_z:
+            intf_z = circuit["uni_z"]["interface_id"]
+            intf_update[intf_z].append(tag_z["value"])
+
+    modi_count = 0
+    intf_details = db.interface_details.find()
+    for document in intf_details:
+        tag_list = intf_update.pop(document["id"], None)
+        if not tag_list:
+            continue
+        result = db.interface_details.update_one(
+            {"id": document["id"]},
+            {"$pull": {"available_vlans": {"$in": tag_list}}}
+        )
+        modi_count += result.modified_count
+
+    added_count = 0
+    for intf_id, tag_list in intf_update.items():
+        tag_list = set(tag_list)
+        new_list = []
+        for tag in range(1, 4096):
+            if tag in tag_list:
+                continue
+            new_list.append(tag)
+        utc_now = datetime.datetime.utcnow()
+        result = db.interface_details.insert_one({
+            "_id": intf_id,
+            "id": intf_id,
+            "inserted_at": utc_now,
+            "updated_at": utc_now,
+            "available_vlans": new_list,
+        })
+        added_count += 1
+    
+    print(f"{modi_count} documents modified. {added_count} documents inserted")
+
+def aggregate_uni_tags(mongo: Mongo):
+    db = mongo.client[mongo.db_name]
+
+    circuits = db.evcs.find(
+        {},
+        {
+            "uni_a": 1,
+            "uni_z": 1
+        }
+    )
+    intf_update = defaultdict(list)
+    for circuit in circuits:
+        tag_a = circuit["uni_a"].get("tag")
+        tag_z = circuit["uni_z"].get("tag")
+        if tag_a:
+            intf_a = circuit["uni_a"]["interface_id"]
+            intf_update[intf_a].append(tag_a["value"])
+        if tag_z:
+            intf_z = circuit["uni_z"]["interface_id"]
+            intf_update[intf_z].append(tag_z["value"])
+
+    if not intf_update:
+        return
+
+    print("Interfaces that probably need their available_vlans modified.")
```

Comment:
Changed the wording here to `Interfaces that are used by UNIs and their tag values`. Not mentioning modification since the script can be used more than 1 time without any problems.
Also detecting conflict from UNI tags with NNI tags. NNI tag usage is detected by looking into `current_path` and `failover_path` for any used tag and the respective interface.

Reply:
Nicely done, Aldo. Looks great.